### PR TITLE
feat(toast): expose Toast wrapper style prop for override

### LINKED ISSUE #4387: Toast is not centered when width exceeds 560px
### Description

When width exceeds 560px, I expect the toast it to still be centered. However, it seems to be aligned off-centered, regardless of width.

### Link to Reproduction

Easier to demonstrate in Chakra docs

### Steps to reproduce

1. Go to https://chakra-ui.com/docs/feedback/toast#custom-component, scroll down to `Custom Component`
2. Change the code to 
```
function CustomToastExample() {
  const toast = useToast({duration: 50000000})
  return (
    <Button
      onClick={() =>
        toast({
          position: "top",
          render: () => (
            <Box color="white" p={3} w="560px" bg="blue.500">
              Hello World
            </Box>
          ),
        })
      }
    >
      Show Toast
    </Button>
  )
}
```
3. Play around with the width that is originally set to `560px`
image

### Chakra UI Version

1.6.4

### Browser

Firefox, Chrome

### Operating System

- [X] macOS
- [ ] Windows
- [ ] Linux

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by karrui**
Closes #4387

This PR is a continuation of #4432 since that PR seems to have been abandoned. As such, the description is copied from that original PR and extended if necessary.

## 📝 Description

When users set a fixed-width on their custom toast, this causes their toast to not be centered. This is because we set a fixed `maxWidth` property on the wrapper component around the toast, which results in the invisible wrapper being centered correctly, but the visible custom component overflowing the parent (the wrapper), making it seem off-centered.

A more detailed explanation of why this arises can be found [here](https://github.com/chakra-ui/chakra-ui/issues/4387#issuecomment-884419618)

## ⛳️ Current behavior (updates)

The div (with `class="chakra-toast__inner"`) wrapping the dispatched `Toast` component currently has a hardcoded style, causing toasts to appear off-center if its child element has a width that is higher than the hardcoded `maxWidth` of `560px`.

## 🚀 New behavior

Instead of only exposing the maxWidth prop so that that behaviour can be fixed, instead opt to expose (a new) `wrapperStyles` prop so users can freely update the hardcoded style and more.

For example, the off-center bug can be fixed with

```ts
// Via instantiation
const toast = useToast({
    position: "top",
    title: "Wrapper style is updated",
    wrapperStyle: {
      maxWidth: "100%",
    },
})

// Via trigger
<Button onClick={() => toast({ wrapperStyle: { maxWidth: "100%" } })}>
    toast
</Button>
```

## 💣 Is this a breaking change (Yes/No):

No

### CODE REVIEW

**Comment by 0x7An** [comment]
+1 for this, overriding styles of toast are very useful.

**Comment by RSchneider94** [comment]
+2 :D are you guys planning to merge this soon? thanks!

**Comment by sozonome** [comment]
+3, really appreciate it if this can be merged soon. 

Currently I'm using `react-hot-toast` when I want full screen toast

If this get merged I can remove that package and use chakra-ui's toast instead

File: .changeset/moody-bees-pay.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@chakra-ui/toast": minor
+---
+
+exposes toast component's wrapper style prop
```
Comment (segunadebayo) [inline .changeset/moody-bees-pay.md:5]:
```suggestion

The `toast` function now exposes a `containerStyle` property you can use to
override the default styles for the toast container.

```jsx live=false
function Example() {
  const toast = useToast()
  return <button>Click me</button>
}
```
```

File: packages/toast/src/toast-manager.tsx [hunk 2]
```diff
@@ -27,7 +27,12 @@ interface Props {
 type CreateToastOptions = Partial<
   Pick<
     ToastOptions,
-    "status" | "duration" | "position" | "id" | "onCloseComplete"
+    | "status"
+    | "duration"
+    | "position"
+    | "id"
+    | "onCloseComplete"
+    | "wrapperStyle"
```
Comment (segunadebayo) [inline packages/toast/src/toast-manager.tsx:35]:
```suggestion
    | "containerStyle"
```

File: packages/toast/src/toast-manager.tsx [hunk 3]
```diff
@@ -167,6 +172,7 @@ export class ToastManager extends React.Component<Props, ToastState> {
       onRequestRemove: () => this.removeToast(String(id), position),
       status: options.status,
       requestClose: false,
+      wrapperStyle: options.wrapperStyle,
```
Comment (segunadebayo) [inline packages/toast/src/toast-manager.tsx:175]:
```suggestion
      containerStyle: options.containerStyle,
```

**Review by segunadebayo — CHANGES_REQUESTED** [review]
Thanks for working on this @karrui. Some feedback to get this merged:

- Please rename all instances of `wrapperStyle` to `containerStyle`, I think it's more intuitive.
- Update the docs to include this feature
- Update the changeset based on my suggestion

**Comment by karrui** [comment]
> Thanks for working on this @karrui. Some feedback to get this merged:
> 
> * Please rename all instances of `wrapperStyle` to `containerStyle`, I think it's more intuitive.
> * Update the docs to include this feature
> * Update the changeset based on my suggestion

Done. Thanks for the review.

**Pull request merged**